libmscore/scoreElement.h

```diff
@@ -407,6 +407,9 @@ class ScoreElement {
       bool isStaffTextBase() const {
             return isStaffText() || isSystemText();
             }
+      bool isVBoxBase() const {
+            return isVBox() || isTBox() || isFBox();
+            }
```

Comment:
I just noticed there is already an isBox() which is defined on line 360 as
```
      bool isBox() const { return isVBox() || isHBox() || isTBox() || isFBox(); }
```
Could you use that instead? Or would the presence of an HBox in the score cause this to not work properly? If you still need it, you might move it closer to the `isBox()` definition. Perhaps even use it in the `isBox()` definition.

Reply:
Only `VBox` and its decendants cause problems. It would be a mistake to exclude `HBox`es from the calculation, since they do exist in continuous view and have an actual width. I can move `isVBox()` just below `isBox()` or use `while (isBox() && !isHBox)` instead. What do you think is best?

Reply:
I kind of like `while (isBox() && !isHBox())`, but I think I would be fine with whatever you decide.

Reply:
I opted to keep `isVBoxBase()` since it makes the logic explicit and I know of at least one other case where there was a bug because not all descendants of `VBox` was handled. The fix handles two of the three `VBox` types, but still misses the last. An explicit function will help tip off developers that they need to maybe handle all elements with base type of `VBox`.

Reply:
I would actually go one step further and redefine `isVBox()` like [this](https://github.com/mattmcclinch/MuseScore/commit/8ec7fa6fafd51d9b070822e3351b60066d184f79). This would have the added benefit of fixing [issue 283546](https://musescore.org/en/node/283546) for free.

Reply:
I am just careful for the case where it is used to know if it is a `VBox` and not a `TBox` or `FBox`. I am not aware of  any such cases, but doing this change may break some other logic later on. It also removes the clean, short way of finding out if something is a `VBox` and not a `TBox` or `FBox`. Anyway, I'm open to be convinced otherwise.

Reply:
VBox seems to be unique in that it is a concrete Element subclass that has subclasses of its own. When I say it is "concrete", I mean it overrides the pure virtual function ScoreElement::type(). If this were to be consistent with the rest of the Element class hierarchy, VBox, TBox, and FBox would all be concrete subclasses of the virtual class VBoxBase. I am not suggesting that you reorganize the class hierarchy in this PR, but I support your decision to stick with isVBoxBase().

Reply:
@mattmcclinch see https://github.com/musescore/MuseScore/blob/master/mscore3.txt#L184
That would suggest that the class hierarchy should anyway be reorganised, sooner rather than later. I am not going to do it in this PR, but it should be done some time.